### DISCUSSION TITLE
How would I get this working with the Uppy client?

### DISCUSSION THREAD

**Discussion by spencerbart** [Klowner/tussle#11] [Q&A]
I'm trying to get this to work on a Cloudflare Worker and I'm following this example https://github.com/Klowner/tussle/tree/main/examples/cloudflare-worker-r2. I am using the Uppy client but it isn't working. 

Here's my Uppy client
```typescript
import Uppy from '@uppy/core';
import { Dashboard } from '@uppy/react';
import Tus from '@uppy/tus';

import '@uppy/core/dist/style.min.css';
import '@uppy/dashboard/dist/style.min.css';

const uppy = new Uppy().use(Tus, { endpoint: 'http://localhost:8787/files'})
export default function Home() {
  return (
    <main>
      <Dashboard uppy={uppy} plugins={["drag-drop"]} />
    </main>
  )
}
```

I must be missing something but I'm not sure what it is.

**Comment by Klowner** [comment]
You're running the example daemon (via miniflare) on your local machine as
in the example, and are there any errors in the browser from Uppy, or any
other errors?

How are you serving the frontend portion? The example bundles the client script into the worker.mjs so it can serve from the same domain, circumventing CORS issues.

I'd also be happy to look at a minimal example project too.

**Reply by spencerbart** [reply]
![Screenshot 2023-05-23 at 13 35 14](https://github.com/Klowner/tussle/assets/38776747/4e87bfc5-127a-4783-b505-27b3a6c62f11)

I just spun up the example in the examples/cloudflare-worker-r2 directory without changing it and this is the error that I get. It's trying to open a directory when it thinks it's a file.

**Reply by Klowner** [reply]
Ah-ha, well it looks like I introduced a bug with the R2 auto merge feature added in 0.7 and the way miniflare stores R2 records.

Until I get a fix published, you can either add `skipMerge: true` to the `TussleStorageR2` constructor in `worker.js` *or* comment out `r2_persist = "./data/r2"` in the `wrangler.toml` within the example directory.
